# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/keyutils/hotfix-gcc4.patch.avr32
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- keyutils-1.6.3/keyutils.h.vanilla	2024-02-06 15:37:41.350515927 +0100
+++ keyutils-1.6.3/keyutils.h	2024-02-06 15:42:25.691538644 +0100
@@ -146,10 +146,10 @@
 struct keyctl_pkey_params {
 	key_serial_t	key_id;		/* Serial no. of public key to use */
 	unsigned int	in_len;		/* Input data size */
-	union {
+	//union {
 		unsigned int	out_len; /* Output buffer size (encrypt/decrypt/sign) */
-		unsigned int	in2_len; /* Second input data size (verify) */
-	};
+		//unsigned int	in2_len; /* Second input data size (verify) */
+	//};
 	unsigned int	__spare[7];
 };
 
--- keyutils-1.6.3/keyutils.c.vanilla	2024-02-06 15:36:28.973510144 +0100
+++ keyutils-1.6.3/keyutils.c	2024-02-06 15:42:44.331540133 +0100
@@ -321,7 +321,7 @@
 	struct keyctl_pkey_params params = {
 		.key_id		= key_id,
 		.in_len		= data_len,
-		.in2_len	= sig_len,
+		.out_len	= sig_len,
 	};
 
 	return keyctl(KEYCTL_PKEY_VERIFY, &params, info, data, sig);
